### ISSUE TITLE
UnfurlUndefined objects causes TypeError during render

### ISSUE THREAD

**Issue by meru1a** [onecommons/unfurl#124]
`UnfurlUndefined` object can end up in a dict that passed to json.dump.

``` UNFURL  WARNING  Configurator render failed
Traceback (most recent call last):
  File "/home/admin/unfurl/unfurl/planrequests.py", line 480, in _render_request
    task.rendered = task.configurator.render(task)
  File "/home/admin/unfurl/unfurl/configurators/terraform.py", line 311, in render
    mainpath, varfilePath = self._prepare_workspace(task, cwd)
  File "/home/admin/unfurl/unfurl/configurators/terraform.py", line 255, in _prepare_workspace
    varpath = self._prepare_vars(task, cwd)
  File "/home/admin/unfurl/unfurl/configurators/terraform.py", line 274, in _prepare_vars
    return cwd.write_file(tfvar [1792 omitted...]    File "/usr/lib/python3.7/json/__init__.py", line 179, in dump
    for chunk in iterable:
  File "/usr/lib/python3.7/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/lib/python3.7/json/encoder.py", line 438, in _iterencode
    o = _default(o)
  File "/usr/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type UnfurlUndefined is not JSON serializable
```

**Comment by AjBreidenbach** [comment]
This seems to happen when a literal Undefined value ends up in the dictionary

**Comment by aszs** [comment]
this should be fixed in 2622e96c7bcd517 -- terraform render now returns early if the task recorded errors
